jmix-flowui/flowui/src/main/java/io/jmix/flowui/impl/DialogsImpl.java

```diff
@@ -879,14 +881,22 @@ public String getHeader() {
 
         @Override
         public BackgroundTaskDialogBuilder<T, V> withText(String text) {
-            this.messageText = text;
+            if (!(content instanceof Span)) {
```

Comment:
if content is a `Span` it doesn't mean that it's a Span that is created for `text`. Let's simplify this code to 

```
Span span = uiComponents.create(Span.class);
span.setText(text);
return withContent(span);
```

So, we don't duplicate code of `withContent` method.